testing/e2e/skr-tester/pkg/command/assert.go

```diff
@@ -156,8 +271,70 @@ func (cmd *AssertCommand) Validate() error {
 	if cmd.admins != nil {
 		count++
 	}
+	if cmd.btpManagerSecretExists {
+		count++
+	}
+	if cmd.editBtpManagerSecret {
+		count++
+	}
+	if cmd.deleteBtpManagerSecret {
+		count++
+	}
 	if count != 1 {
-		return errors.New("only one of machineType, clusterOIDCConfig, kubeconfigOIDCConfig, or admins must be specified")
+		return errors.New("you must use exactly one of machineType, clusterOIDCConfig, kubeconfigOIDCConfig, admins, btpManagerSecretExists, editBtpManagerSecret, or deleteBtpManagerSecret")
+	}
+	return nil
+}
+
+func (cmd *AssertCommand) checkBTPManagerSecret(kubeconfig []byte) error {
+	restCfg, err := clientcmd.RESTConfigFromKubeConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("while creating rest config from kubeconfig: %w", err)
+	}
+	k8sCli, err := client.New(restCfg, client.Options{
+		Scheme: scheme.Scheme,
+	})
+	if err != nil {
+		return fmt.Errorf("while creating k8s client: %w", err)
+	}
+	secrets := &v1.SecretList{}
+	listOpts := []client.ListOption{
+		client.InNamespace("kyma-system"),
+		client.MatchingFields{"metadata.name": "sap-btp-manager"},
+	}
+	err = k8sCli.List(context.Background(), secrets, listOpts...)
+	if err != nil {
+		return fmt.Errorf("while getting secret from instance: %w", err)
+	}
+
+	if len(secrets.Items) != 1 {
+		return fmt.Errorf("found %d secrets but expected 1", len(secrets.Items))
+	}
+	secret := secrets.Items[0]
+	if secret.Labels["app.kubernetes.io/managed-by"] != "kcp-kyma-environment-broker" {
+		return fmt.Errorf("secret label 'app.kubernetes.io/managed-by' is not 'kcp-kyma-environment-broker'")
+	}
+	fmt.Println("BTP manager secret exists")
+
+	requiredKeys := []string{"clientid", "clientsecret", "sm_url", "tokenurl", "cluster_id"}
+	for _, key := range requiredKeys {
+		if _, exists := secret.Data[key]; !exists {
+			return fmt.Errorf("secret data key %s not found", key)
+		}
+	}
+	fmt.Println("Required keys exist in BTP manager secret")
+
+	expectedCreds := map[string]string{
```

Comment:
We could define two maps in one place: 
- map with default/expected values
- map with "edited" values
Then it is easier to see if the values differ.
But this is nit-picking.